frontend/src/components/Room.tsx

```diff
@@ -1,179 +1,326 @@
-      end: end,
-      course_name: { pl: "REZERWACJA" },
-      id: 0,
+
+    // przechowuje eventy do wyświetlenia w kalendarzu
+    const roomActivities = mapActivitiesToEvents(activities);
+
+    const pushNewRoomAnnotation = useMutation((newAnnotation) => {
+        const roomData = {
+            roomAnnotation: newAnnotation,
+        };
+
+        return Api.authApi.patch(`room/update/${id}`, roomData, {
+            headers: {
+                Authorization: `Bearer ${token}`,
+            },
+        });
     });
-  };
-
-  return (
-    <div className=" mx-16 py-10">
-      <div className="flex flex-col items-center text-center">
-        <h1 className="text-3xl font-bold mb-4">Sala</h1>
-        {isRoomFetching ? (
-          <p className="text-center">Ładowanie danych...</p>
-        ) : (
-          <div className="w-60 text-start">
-            <p className="pb-4 text-4xl text-center font-bold">
-              {room?.number}
-            </p>
-
-            {!token ? (
-              <p className="mb-2 text-lg">
-                <span className="font-bold">Rodzaj: </span>
-                {room?.roomAnnotation}
-              </p>
-            ) : (
-              <div className="mb-2">
-                <label className=" text-lg font-bold">
-                  Rodzaj:
-                  <select
-                    value={
-                      Object.entries(RoomAnnotation).find(
-                        ([key, val]) => val === roomAnnotation
-                      )?.[0]
+
+    const roomAnnotationChange = (e) => {
+        e.preventDefault();
+        pushNewRoomAnnotation.mutate(RoomAnnotation[e.target.value as keyof typeof RoomAnnotation]); // e.target.value)
+        setRoomAnnotation(e.target.value);
+    };
+
+    const handleSelectSlot = ({start, end}: { start: Date; end: Date }) => {
+        setBackgroundEvent({
+            start: start, end: end, course_name: {pl: "REZERWACJA"}, id: 0,
+        });
+    };
+
+    const [cbxDeleteRangeChecked, setCbxDeleteRangeChecked] = React.useState(false);
+    const cbxDeleteRangeChanged = () => {
+        setCbxDeleteRangeChecked(!cbxDeleteRangeChecked);
+    };
+
+    const [cbxDeleteDateChecked, setCbxDeleteDateChecked] = React.useState(false);
+    const cbxDeleteDateChanged = () => {
+        setCbxDeleteDateChecked(!cbxDeleteDateChecked);
+    };
+
+    const [cbxSelectDateChecked, setCbxSelectDateChecked] = React.useState(false);
+    const cbxSelectDateChanged = () => {
+        setCbxSelectDateChecked(!cbxSelectDateChecked);
+    };
+
+    const [smallCalendarValues, setSmallCalendarValues] = useState([new Date(), new Date()]);
+    const smallCalendarChanged = (value: React.SetStateAction<Date[]>, _: any) => {
+        setSmallCalendarValues(value);
+        setTimeout(() => {
+            window.scrollTo(0, document.body.scrollHeight);
+        })
+    }
+
+    const calendarRef = useRef();
+
+    const CalendarToPrint = React.forwardRef((props: any, ref) => {
```

Comment:
can't you just move this to other component and only render it here? because it seems like this is component created in component, in which case you can just move it to other file and import it